unicon_backend/evaluator/tasks/programming/steps.py

```diff
@@ -279,6 +287,36 @@ def _serialize_data(data: str | int | float | bool) -> str:
         return program
 
 
+class OutputStep(Step):
+    subgraph_socket_ids: ClassVar[set[str]] = set()
+
+    @model_validator(mode="after")
+    def check_non_empty_outputs(self) -> Self:
+        if len(self.inputs) == 0:
+            raise ValueError("Output step must have at least one input")
+
+        return self
+
+    def run(self, var_inputs: dict[SocketName, ProgramVariable], *_) -> Program:
+        program: Program = [*self.debug_stmts()]
+
+        program.append("import json")
+        result = (
+            "{"
+            + ", ".join(
+                (
+                    f'"{key}": {variable_name}'
+                    for key, variable_name in var_inputs.items()
+                    if not key.startswith("CONTROL")
```

Comment:
wdyt of using output sockets instead of `var_inputs` here so that its semantically clearer we are using outputs but getting values from `var_inputs`

Reply:
i am confused what this means: does this mean and output sockets and input sockets should be identical?

Reply:
instead of looping over `var_inputs`, we iterate through output socket names and then access values from `var_inputs`

Reply:
there is a current check that forces input socket + output socket to be unique that would block that - is that safe to delete?

Reply:
The sockets should still be unique. I just thought it would be clearer if we do something like

```py
[ 
  var_inputs[out_socket.id] # or var_inputs.get(out_socket.id)
  for out_socket in self.outputs
]
```

But this isn't a big deal so I'm good to merge this